### PR TITLE
fix: reduce test run time

### PR DESCRIPTION
**Pull request by adodon2go**
**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            cd $GITHUB_WORKSPACE
+            go mod download
+      - uses: ./.github/actions/check-diskspace
+      - name: run zot development-mode unit tests (possibly using failure injection)
+        run: make test-devmode
+      - name: upload coverage
+        uses: actions/upload-artifact@v3
+        with:
+          name: coverage-devmode
+          path: coverage-dev-minimal.txt
```
Comment (andaaron) [inline .github/workflows/test.yaml:132]: Maybe use a single wildcard expression to upload all these files in a single action. Would that work?

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
+# zot [![build](https://github.com/project-zot/zot/actions/workflows/ci-cd.yml/badge.svg?branch=main)](https://github.com/project-zot/zot/actions/workflows/ci-cd.yml) [![test](https://github.com/project-zot/zot/actions/workflows/test.yml/badge.svg?branch=main)](https://github.com/project-zot/zot/actions/workflows/test.yml) [![codecov.io](http://codecov.io/github/project-zot/zot/coverage.svg?branch=main)](http://codecov.io/github/project-zot/zot?branch=main) [![Conformance Results](https://github.com/project-zot/zot/workflows/conformance/badge.svg)](https://github.com/project-zot/zot/actions?query=workflow%3Aconformance) [![CodeQL](https://github.com/project-zot/zot/workflows/CodeQL/badge.svg)](https://github.com/project-zot/zot/actions?query=workflow%3ACodeQL) [![CII Best Practices](https://bestpractices.coreinfrastructure.org/projects/5425/badge)](https://bestpractices.coreinfrastructure.org/projects/5425) [![Go Reference](https://pkg.go.dev/badge/zotregistry.io/zot.svg)](https://pkg.go.dev/zotregistry.io/zot)
```
Comment (andaaron) [inline README.md:1]:
```suggestion
# zot [![build](https://github.com/project-zot/zot/actions/workflows/ci-cd.yml/badge.svg?branch=main)](https://github.com/project-zot/zot/actions/workflows/ci-cd.yml) [![test](https://github.com/project-zot/zot/actions/workflows/test.yaml/badge.svg?branch=main)](https://github.com/project-zot/zot/actions/workflows/test.yaml) [![codecov.io](http://codecov.io/github/project-zot/zot/coverage.svg?branch=main)](http://codecov.io/github/project-zot/zot?branch=main) [![Conformance Results](https://github.com/project-zot/zot/workflows/conformance/badge.svg)](https://github.com/project-zot/zot/actions?query=workflow%3Aconformance) [![CodeQL](https://github.com/project-zot/zot/workflows/CodeQL/badge.svg)](https://github.com/project-zot/zot/actions?query=workflow%3ACodeQL) [![CII Best Practices](https://bestpractices.coreinfrastructure.org/projects/5425/badge)](https://bestpractices.coreinfrastructure.org/projects/5425) [![Go Reference](https://pkg.go.dev/badge/zotregistry.io/zot.svg)](https://pkg.go.dev/zotregistry.io/zot)
```
File name is `test.yaml` in the repo and `test.yml` here

**Review by andaaron — COMMENTED** [review]
Overall a very good propsal

File: .github/workflows/test.yaml (hunk 1 above)
Comment (adodon2go) [reply .github/workflows/test.yaml:131]: This doesn't work. Also specifying multiple files in 'path' field does not work

File: .github/ISSUE_TEMPLATE/feature_request.yaml
Comment (andaaron) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:1]: Please test these templates continue to work with the new names.

File: .github/dependabot.yaml
Comment (andaaron) [inline .github/dependabot.yaml:1]: I think this one needs to stay .yml. Thats' the name mentioned in the documentation I found.

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+      - name: run zot privileged tests
+        run: sudo env "PATH=$PATH" make privileged-test
+      - name: upload coverage
+        uses: actions/upload-artifact@v3
+        with:
+          name: coverage-devmode
+          path: coverage-dev-needprivileges.txt
+      - uses: ./.github/actions/check-diskspace
+  test-coverage:
+    name: generate test coverage
```
Comment (andaaron) [inline .github/workflows/test.yaml:146]:
```suggestion
    name: Generate test coverage
```

To be consistent with the other job names in the same file

File: .github/workflows/codeql-analysis.yaml
[the remark this answers is not in the document]
Comment (adodon2go) [reply .github/workflows/codeql-analysis.yaml:1]: It actually run and to 18 min to complete. 
Do you mean this warning?

> [Analyze (go)](https://github.com/project-zot/zot/actions/runs/6291972459/job/17080894262#step:5:239)
> Passing the GOFLAGS env parameter to the init action is deprecated. Please move this to the analyze action.

[the remark this answers is not in the document]
Comment (andaaron) [reply .github/workflows/codeql-analysis.yaml:1]: I was referring to this: https://github.com/project-zot/zot/pull/1832/checks?check_run_id=17081035115
Now I see there's also your link. Do we have 2 separate jobs?

File: .github/ISSUE_TEMPLATE/feature_request.yaml
Comment (adodon2go) [reply .github/ISSUE_TEMPLATE/feature_request.yaml:1]: Can't test this on my fork. Should be tested when PR is merged to main

File: .github/workflows/codeql-analysis.yaml
[the remark this answers is not in the document]
Comment (andaaron) [reply .github/workflows/codeql-analysis.yaml:1]: Looks like there is a leftover check from the other file, but we also have a "new" check. Resolving this.

**Review by andaaron — DISMISSED** [review]

**Review by andaaron — DISMISSED** [review]

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+      - name: run zot privileged tests
+        run: sudo env "PATH=$PATH" make privileged-test
+      - name: upload coverage
+        uses: actions/upload-artifact@v3
+        with:
+          name: coverage-devmode
+          path: coverage-dev-*.txt
+      - uses: ./.github/actions/check-diskspace
+  test-coverage:
+    name: Centralize test coverage
```
Comment (rchincha) [inline .github/workflows/test.yaml:136]: "Collect all..."

**Comment by rchincha** [comment]
Build parts and test parts need to be mandatory

**Comment by rchincha** [comment]
codeql-analysys.yaml -> .yml

**Review by andaaron — APPROVED** [review]

**Comment by rchincha** [comment]
Build and test ZOT (darwin, amd64) Expected — Waiting for status to be reported
Required
Build and test ZOT (linux, amd64) Expected — Waiting for status to be reported
Required

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**